### PR TITLE
RS-210: Guides for data ingestion and querying

### PR DESCRIPTION
**Pull request by atimin**
* Change "Developers" menu item
* Fix code snippet on the landing page
* Write data ingestion and querying guides

### CODE REVIEW

File: docs/guides/data-ingestion.mdx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+In case of little size records, ReductStore provides a batch ingestion mode. In this mode, the server receives a batch of records in a single request to reduce the overhead of the HTTP protocol.
+
+
+### Limitations
+
+The following limitations are applied to the data ingestion process:
+
+* Currently, records and their metadata are immutable. Once a record is ingested successfully, it cannot be overwritten or changed.
+* ReductStore doesn't have any limitations on the size of the record, however, the metadata is sent in the HTTP headers of the request and the size of the headers can be limited by the server or client configuration.
+* Since ReaductStore pre-allocate space for records and store them together in a block, the size must be known in advance even though the record is sent in chunks.
```
Comment (AnthonyCvn) [inline docs/guides/data-ingestion.mdx:41]: I got confused with the "block size" like for the parameter "max block size", but you meant the size of the blob that need to be stored?

Like: "ReaductStore needs to know the total size of a record in advance, even if it's sent in chunks. This allows it to reserve the right amount of space for optimal performance."

File: docs/guides/data-ingestion.mdx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            {SimpleCurl}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
+
+### Streaming Data
+
+For large records, using the streaming approach is recommended. Here, a client application sends a record in chunks, avoiding loading the entire record into memory. This approach is particularly beneficial for real-time data ingestion, where data is received from a stream and needs to be sent to ReductStore without buffering.
+
+:::info
```
Comment (AnthonyCvn) [inline docs/guides/data-ingestion.mdx:92]: can be a warning

File: docs/guides/data-quering.mdx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+# Data Querying
+
+## Concepts
+
+ReducStore provides an efficient data retrieval solution by batching multiple records within a specified time interval into a single HTTP request, which is beneficial for managing large volumes of data as it reduces the number of requests and overall delay.
+
+The query process is designed as an iterator, returning a batch of records in each iteration. This method allows data to be processed in segments, an approach that is useful when managing large datasets.
+
+While it is possible to retrieve a record by its timestamp, this method is less optimal than querying by a time range due to the lack of batching. However, this approach can be used when timestamps serve as identifiers for non-time series records, such as models or configurations.
```
Comment (AnthonyCvn) [inline docs/guides/data-quering.mdx:23]: suggestion: "However, this approach can be useful for querying specific versions of non-time series records, such as AI models, configurations, or file versions, when timestamps are used as identifiers."

File: docs/guides/data-quering.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+## Query Parameters
+
+Data can be queried using the ReductStore CLI, SDKs or HTTP API. The query parameters are the same for all interfaces and include
```
Comment (AnthonyCvn) [inline docs/guides/data-quering.mdx:27]: missing colons before list

File: docs/guides/data-ingestion.mdx (hunk 2 above)
Comment (atimin) [reply docs/guides/data-ingestion.mdx:92]: it isn't dangerous or something like this. I used warning for cases where data can be lost.

**Pull request merged**